### PR TITLE
Add changelog

### PR DESCRIPTION
**Pull request by refi93**
Motivation: There is currently no formal documentation capturing changes between different versions of this library, therefore external developers lack the context of when switching to a new version is actually needed and what are the relevant changes.

Changes:
* add a changelog file capturing most relevant changes so far (since version 2.0.0 as there is no point in keeping track of older versions that do not work anymore as they were meant for Byron era)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+Works with Ledger Cardano app 2.2.0 and is backwards compatible with older versions down to 2.0.4/2.0.5*. Older versions of this js library do not support Ledger Cardano app 2.2.0, hence an update to this version of the library is required before Ledger Cardano app 2.2.0 is released.
+
+### Added
+
+- Support for Allegra-era transaction min validity property, transaction TTL is now optional: https://github.com/vacuumlabs/ledgerjs-cardano-shelley/pull/27/files
+- Support for Mary-era multiasset outputs https://github.com/vacuumlabs/ledgerjs-cardano-shelley/pull/27/files
+- Update docs with a stake pool registration (as an owner) example https://github.com/vacuumlabs/ledgerjs-cardano-shelley/pull/23
+
+ 
+### Changed
```
Comment (janmazak) [inline CHANGELOG.md:18]: Account numbers now in line with Trezor.

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+### Fixed
+
+- Fixed incorrect validation of numerical parameters passed as strings which failed for values over JS max safe integer: https://github.com/vacuumlabs/ledgerjs-cardano-shelley/pull/29
+
+## [2.1.0] - [December 11th 2020]
+
+Works with Ledger Cardano app 2.1.0 and is backwards compatible with 2.0.4/2.0.5 as well*. Older versions of this library do not support Ledger Cardano app 2.1.0, therefore an update to this version of the library is required before Ledger Cardano app 2.1.0 is released.
```
Comment (janmazak) [inline CHANGELOG.md:26]: The asterisk gets lost in markdown formatting.

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+and this project adheres to [Semantic Versioning](http://semver.org/).
+
+## [2.2.0] - [February 8th 2020]
+
+Works with Ledger Cardano app 2.2.0 and is backwards compatible with older versions down to 2.0.4/2.0.5*. Older versions of this js library do not support Ledger Cardano app 2.2.0, hence an update to this version of the library is required before Ledger Cardano app 2.2.0 is released.
+
+### Added
+
+- Support for Allegra-era transaction min validity property, transaction TTL is now optional: https://github.com/vacuumlabs/ledgerjs-cardano-shelley/pull/27/files
+- Support for Mary-era multiasset outputs https://github.com/vacuumlabs/ledgerjs-cardano-shelley/pull/27/files
```
Comment (janmazak) [inline CHANGELOG.md:14]: No minting yet, perhaps worth mentioning explicitly.

**Review by janmazak — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (janmazak) [reply CHANGELOG.md:18]: And bech32 pool id.

Comment (refi93) [reply CHANGELOG.md:18]: those are not changes that depend on ledgerjs (in fact, different versions of the app will behave differently in that regard with the same version of ledgerjs) and IMHO a similar changelog should exist for the ledger cardano app which could be referenced from here. So I would propose once we go ahead with this js change log, to replicate it for the cardano app.

Comment (ppershing) [reply CHANGELOG.md:18]: agreed. We should do some tidyup in firmware app docs as well

File: README.md [hunk 4]
```diff
@@ -17,3 +17,8 @@ Automated tests are provided. There are two types of tests
 
 1. `yarn test-integration`. Tests JS api.
 2. `yarn test-direct`. Mostly tests vartious corner cases. There are some extensive tests which are disabled by default, see tests source code. Also note that for these tests it is advised to install developer version of Cardano app with _headless_ mode enabled, otherwise you spend your entire life confirming various prompts on device.
```
Comment (janmazak) [inline README.md:19]: I think headless mode should be suggested for all tests, including the integration ones.

**Review by janmazak — APPROVED** [review]
